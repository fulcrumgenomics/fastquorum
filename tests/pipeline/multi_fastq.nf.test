nextflow_pipeline {

    name "Test workflow - Multiple FASTQs"
    script "main.nf"
    tag "multi_fastq"
    tag "pipeline"

    test("Multiple FASTQs - R&D mode") {

        when {
            params {
                input  = 'https://raw.githubusercontent.com/nf-core/test-datasets/fastquorum/testdata/samplesheets/samplesheet.multi_fastq.csv'
                fasta  = 'https://github.com/nf-core/test-datasets/raw/fastquorum/references/chr17.fa'
                outdir = "$outputDir"
                mode   = "rd"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(UTILS.removeNextflowVersion("$outputDir")).match("multi_fastq_rd_software_versions") },
                { assert workflow.trace.succeeded().size() == 28 },
                { assert new File("$outputDir/multiqc/multiqc_report.html").exists() },
                { assert new File("$outputDir/consensus_filtering/filtered/SRR6109255_two_fastqs/SRR6109255_two_fastqs.cons.filtered.bam").exists() },
                { assert new File("$outputDir/consensus_filtering/filtered/SRR6109255_three_fastqs/SRR6109255_three_fastqs.cons.filtered.bam").exists() },
                { assert new File("$outputDir/consensus_filtering/filtered/SRR6109255_four_fastqs/SRR6109255_four_fastqs.cons.filtered.bam").exists() },
                { assert snapshot(
                    path("$outputDir/grouping/groupreadsbyumi/SRR6109255_two_fastqs/SRR6109255_two_fastqs.grouped-family-sizes.txt"),
                    path("$outputDir/metrics/duplex_seq/SRR6109255_two_fastqs/SRR6109255_two_fastqs.duplex_seq_metrics.family_sizes.txt"),
                    path("$outputDir/grouping/groupreadsbyumi/SRR6109255_three_fastqs/SRR6109255_three_fastqs.grouped-family-sizes.txt"),
                    path("$outputDir/metrics/duplex_seq/SRR6109255_three_fastqs/SRR6109255_three_fastqs.duplex_seq_metrics.family_sizes.txt"),
                    path("$outputDir/grouping/groupreadsbyumi/SRR6109255_four_fastqs/SRR6109255_four_fastqs.grouped-family-sizes.txt"),
                    path("$outputDir/metrics/duplex_seq/SRR6109255_four_fastqs/SRR6109255_four_fastqs.duplex_seq_metrics.family_sizes.txt"),
                ).match("multi_fastq_rd") }
            )
        }

    }

    test("Multiple FASTQs - high throughput mode") {

        when {
            params {
                input  = 'https://raw.githubusercontent.com/nf-core/test-datasets/fastquorum/testdata/samplesheets/samplesheet.multi_fastq.csv'
                fasta  = 'https://github.com/nf-core/test-datasets/raw/fastquorum/references/chr17.fa'
                outdir = "$outputDir"
                mode   = "ht"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(UTILS.removeNextflowVersion("$outputDir")).match("multi_fastq_ht_software_versions") },
                { assert workflow.trace.succeeded().size() == 25 },
                { assert new File("$outputDir/multiqc/multiqc_report.html").exists() },
                { assert new File("$outputDir/consensus_filtering/filtered/SRR6109255_two_fastqs/SRR6109255_two_fastqs.cons.filtered.bam").exists() },
                { assert new File("$outputDir/consensus_filtering/filtered/SRR6109255_three_fastqs/SRR6109255_three_fastqs.cons.filtered.bam").exists() },
                { assert new File("$outputDir/consensus_filtering/filtered/SRR6109255_four_fastqs/SRR6109255_four_fastqs.cons.filtered.bam").exists() },
                { assert snapshot(
                    path("$outputDir/grouping/groupreadsbyumi/SRR6109255_two_fastqs/SRR6109255_two_fastqs.grouped-family-sizes.txt"),
                    path("$outputDir/metrics/duplex_seq/SRR6109255_two_fastqs/SRR6109255_two_fastqs.duplex_seq_metrics.family_sizes.txt"),
                    path("$outputDir/grouping/groupreadsbyumi/SRR6109255_three_fastqs/SRR6109255_three_fastqs.grouped-family-sizes.txt"),
                    path("$outputDir/metrics/duplex_seq/SRR6109255_three_fastqs/SRR6109255_three_fastqs.duplex_seq_metrics.family_sizes.txt"),
                    path("$outputDir/grouping/groupreadsbyumi/SRR6109255_four_fastqs/SRR6109255_four_fastqs.grouped-family-sizes.txt"),
                    path("$outputDir/metrics/duplex_seq/SRR6109255_four_fastqs/SRR6109255_four_fastqs.duplex_seq_metrics.family_sizes.txt"),
                ).match("multi_fastq_ht") }
            )
        }

    }

}
