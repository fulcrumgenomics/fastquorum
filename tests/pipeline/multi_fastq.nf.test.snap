{
    "multi_fastq_rd": {
        "content": [
            "SRR6109255_two_fastqs.grouped-family-sizes.txt:md5,d2b8b8eb6c01a58f9aea6596b44e3735",
            "SRR6109255_two_fastqs.duplex_seq_metrics.family_sizes.txt:md5,1439dc926491ef89f34dc5c56dc61b56",
            "SRR6109255_three_fastqs.grouped-family-sizes.txt:md5,d2b8b8eb6c01a58f9aea6596b44e3735",
            "SRR6109255_three_fastqs.duplex_seq_metrics.family_sizes.txt:md5,1439dc926491ef89f34dc5c56dc61b56",
            "SRR6109255_four_fastqs.grouped-family-sizes.txt:md5,150a851a1ae78822b65b74040d05172c",
            "SRR6109255_four_fastqs.duplex_seq_metrics.family_sizes.txt:md5,60dafa16823744a37fe9ad85d08b038c"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-20T01:50:53.564089"
    },
    "multi_fastq_ht_software_versions": {
        "content": [
            "{ALIGN_CONSENSUS_BAM={bwa=0.7.17-r1188, fgbio=2.0.2, samtools=1.16.1}, ALIGN_RAW_BAM={bwa=0.7.17-r1188, fgbio=2.0.2, samtools=1.16.1}, CALLANDFILTERDUPLEXCONSENSUSREADS={fgbio=2.0.2}, COLLECTDUPLEXSEQMETRICS={fgbio=2.0.2}, FASTQC={fastqc=0.12.1}, FASTQTOBAM={fgbio=2.0.2}, GROUPREADSBYUMI={fgbio=2.0.2}, Workflow={nf-core/fastquorum=v1.0dev}}"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-20T01:58:10.226349"
    },
    "multi_fastq_rd_software_versions": {
        "content": [
            "{ALIGN_CONSENSUS_BAM={bwa=0.7.17-r1188, fgbio=2.0.2, samtools=1.16.1}, ALIGN_RAW_BAM={bwa=0.7.17-r1188, fgbio=2.0.2, samtools=1.16.1}, CALLDDUPLEXCONSENSUSREADS={fgbio=2.0.2}, COLLECTDUPLEXSEQMETRICS={fgbio=2.0.2}, FASTQC={fastqc=0.12.1}, FASTQTOBAM={fgbio=2.0.2}, FILTERCONSENSUSREADS={fgbio=2.0.2}, GROUPREADSBYUMI={fgbio=2.0.2}, Workflow={nf-core/fastquorum=v1.0dev}}"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-20T01:50:53.548509"
    },
    "multi_fastq_ht": {
        "content": [
            "SRR6109255_two_fastqs.grouped-family-sizes.txt:md5,d2b8b8eb6c01a58f9aea6596b44e3735",
            "SRR6109255_two_fastqs.duplex_seq_metrics.family_sizes.txt:md5,1439dc926491ef89f34dc5c56dc61b56",
            "SRR6109255_three_fastqs.grouped-family-sizes.txt:md5,d2b8b8eb6c01a58f9aea6596b44e3735",
            "SRR6109255_three_fastqs.duplex_seq_metrics.family_sizes.txt:md5,1439dc926491ef89f34dc5c56dc61b56",
            "SRR6109255_four_fastqs.grouped-family-sizes.txt:md5,150a851a1ae78822b65b74040d05172c",
            "SRR6109255_four_fastqs.duplex_seq_metrics.family_sizes.txt:md5,60dafa16823744a37fe9ad85d08b038c"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-20T01:58:10.241449"
    }
}